### ISSUE TITLE
Ruby content.meta does not correctly apply X-Riak-Meta- prefix [JIRA: CLIENTS-1109]

### ISSUE THREAD

**Issue by mikedotson** [basho/riak-ruby-client#312]
Please ensure the following information is supplied for new issues:

- [ ] Riak Ruby Client version
riak-client (2.6.0)
- [ ] Ruby version
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
- [ ] Riak version
2.0.8
- [ ] Operating System / Distribution & Version
Distributor ID:	Ubuntu
Description:	Ubuntu 16.10
Release:	16.10
Codename:	yakkety
- [ ] Riak `error.log` file, if applicable
N/A
- [ ] What methods were being executed during the error
content.meta['Version'] = 1.0
puts #{content.meta}
- [ ] What you *expected* to have happen
Object should have contained header: `X-Riak-Meta-Version: 1.0` instead it contained header of `Version: 1.0`.  Retrieving, the hash contained Value.  However, when object had X-Riak-Meta-Version, the hash contained the key `X-Riak-Meta-Version`
- [ ] Sample data, if applicable
Create first key using ruby client:
```
#!/usr/bin/ruby

require 'riak'

client=Riak::Client.new host: '10.10.10.101', pb_port: 8087

bucket=client.bucket 'basho'


ScreenName=bucket.new 'bashoman1'
ScreenName.data='test'
ScreenName.content.meta['Version'] = 1.0
ScreenName.store
```
Create Second key using Curl:
```
curl -XPUT http://10.10.10.101:8098/buckets/basho/keys/bashoman2 -H "Content-Type: text/plain" -H "X-Riak-Meta-Version: 2.0" -d "my awesome data"
```
Create Third key using Curl:
```
curl -XPUT http://10.10.10.101:8098/buckets/basho/keys/bashoman3 -H "Content-Type: text/plain" -H "Version: 1.0" -d "my awesome data" 
```

Retrieve keys using ruby:
```
#!/usr/bin/ruby
require 'riak'
client=Riak::Client.new host: '10.10.10.101', pb_port: 8087

bucket=client.bucket 'basho'

ScreenName=bucket.new 'bashoman'
bashoman1 = ScreenName.bucket.get("bashoman1")
bashoman2 = ScreenName.bucket.get("bashoman2")
bashoman3 = ScreenName.bucket.get("bashoman3")
puts "Bashoman1 Metadata: #{bashoman1.content.meta}"
puts "Bashoman2 Metadata: #{bashoman2.content.meta}"
puts "Bashoman3 Metadata: #{bashoman3.content.meta}"
```

Output is as follows:
```
Bashoman1 Metadata: {"Version"=>"1.0"}
Bashoman2 Metadata: {"X-Riak-Meta-Version"=>"2.0"}
Bashoman3 Metadata: {}
```

Header information for each key using wget -S:
```
wget -q -O /dev/null -S http://10.10.10.101:8098/buckets/basho/keys/bashoman1
  HTTP/1.1 200 OK
  X-Riak-Vclock: a85hYGBgzGDKBVIcmkzt2rPnxVpmMCUy5rEy3JkaeYkvCwA=
  Version: 1.0
  Vary: Accept-Encoding
  Server: MochiWeb/1.1 WebMachine/1.10.6 (no drinks)
  Link: </buckets/basho>; rel="up"
  Last-Modified: Wed, 05 Apr 2017 16:22:52 GMT
  ETag: "77g2gXDqSdHLjOfOz8lFBj"
  Date: Wed, 05 Apr 2017 16:24:06 GMT
  Content-Type: application/json; charset=UTF-8
  Content-Length: 6

wget -q -O /dev/null -S http://10.10.10.101:8098/buckets/basho/keys/bashoman2
  HTTP/1.1 200 OK
  X-Riak-Vclock: a85hYGBgzGDKBVIcmkzt2l1l879mMCUy5rEyPJ8aeYkvCwA=
  X-Riak-Meta-Version: 2.0
  Vary: Accept-Encoding
  Server: MochiWeb/1.1 WebMachine/1.10.6 (no drinks)
  Link: </buckets/basho>; rel="up"
  Last-Modified: Wed, 05 Apr 2017 16:23:03 GMT
  ETag: "5T6sfsYW8VrxKh6xnor073"
  Date: Wed, 05 Apr 2017 16:24:31 GMT
  Content-Type: text/plain
  Content-Length: 15

wget -q -O /dev/null -S http://10.10.10.101:8098/buckets/basho/keys/bashoman3
  HTTP/1.1 200 OK
  X-Riak-Vclock: a85hYGBgzGDKBVIcmkzt2l2haWczmBIZ81gZ3kyNvMSXBQA=
  Vary: Accept-Encoding
  Server: MochiWeb/1.1 WebMachine/1.10.6 (no drinks)
  Link: </buckets/basho>; rel="up"
  Last-Modified: Wed, 05 Apr 2017 16:23:08 GMT
  ETag: "7SOgfUiCpMKGEGMpuRalFr"
  Date: Wed, 05 Apr 2017 16:24:45 GMT
  Content-Type: text/plain
  Content-Length: 15
```

Per [Documentation](http://www.rubydoc.info/gems/riak-client/Riak/RContent#meta=-instance_method), setting `Version` should have resulted in an object with a `X-Riak-Meta-Version` header as is seen in `bashoman2`.  Retrieving the metadata for `bashoman2` should have resulted in the meta key hash as is seen with `bashoman1`, `{"Version"=>"1.0"}`

**Comment by lukebakken** [comment]
Best bug report ever! Thanks!

**Comment by lukebakken** [comment]
The only place in the Ruby client code that `X-Riak-Meta-*` is specially handled is [here](https://github.com/basho/riak-ruby-client/blob/master/lib/riak/rcontent.rb#L177-L196). This is when an object is loaded from map/reduce. I don't see special handling anywhere else in the Ruby code, nor in the Java or Python clients. I am searching the Riak code itself to see how `X-Riak-Meta-*` is handled, both by the HTTP and PB interfaces.

**Comment by lukebakken** [comment]
@mikedotson - Since the Ruby client is protobuf-only, we send user metadata in [the `usermeta`](https://github.com/basho/riak_pb/blob/develop/src/riak_kv.proto#L234) field of `RpbContent`. Riak just stuffs the data from here into the object it stores. There are no headers involved, because it's protobuf.

When an HTTP client posts data to Riak's HTTP API, the only way Riak "knows" that the header is for user metadata is via the `X-Riak-Meta-` header. So, when you post data via HTTP using that header, Riak stores it in the user metadata part of the object, and also uses that header on HTTP retrieval of that object.

As you can see in your examples, Riak doesn't prepend `X-Riak-Meta-` to the `Version` value when it is retrieved via the HTTP interface. That is because the value stored with the Riak object is what is passed in - `Version` through the protobuf store operation, `X-Riak-Meta-Version` via the HTTP store operation.

In short, if you want to have the `X-Riak-Meta-` prefix on an HTTP or PB get, you must store it with that prefix from your client.
